src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
+
+    Parameters
+    ----------
+
+    len_x : float
+        Length of the row dimension of the vertex grid
+
+    len_y : float
+        Length of the column dimension of the vertex grid
+
+    Returns
+    -------
+
+    face_indices : np.array
+        (n, 3) array defining faces of the triangulated mesh for the DEM.
+    """
+    x = np.array(range(len_x - 1))
+    y = np.array(range(len_y - 1))
+    xg, yg = np.meshgrid(x, y)
+    vertex_sw = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x + 1, y + 1)
+    vertex_ne = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x, y + 1)
+    vertex_nw = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x + 1, y)
+    vertex_se = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    face_indices_1 = np.array((vertex_ne, vertex_sw, vertex_nw)).transpose()
+    face_indices_2 = np.array((vertex_se, vertex_sw, vertex_ne)).transpose()
+    face_indices = np.concatenate((face_indices_1, face_indices_2))
+
+    return face_indices
+
+
+def _transform_vertices_spherical_shell(vertices, planet_radius):
+    """
+    Convert the (lat, lon, elevation) vertices from the initial vertex generation
+    into (x, y, z) values for spherical shell atmosphere geometries.
+    """
+    planet_radius_m = planet_radius.m_as(ucc.get("length"))
+    theta_r = np.deg2rad(90 - vertices[:, 0])
+    phi_r = np.deg2rad(vertices[:, 1])
+    x = np.sin(theta_r) * np.cos(phi_r) * (planet_radius_m + vertices[:, 2])
+    y = np.sin(theta_r) * np.sin(phi_r) * (planet_radius_m + vertices[:, 2])
+    z = np.cos(theta_r) * (planet_radius_m + vertices[:, 2])
+    vertices_new = np.array((x, y, z)).transpose()
+
+    return vertices_new
+
+
+def _mean_coordinates(vertices):
+    """
+    Return the mean of the latitude and longitude values in the vertex list.
+    The mean is computed as the average of the smallest and largest value in each set.
+    """
+    (theta_min, theta_max), (phi_min, phi_max) = _minmax_coordinates(vertices)
+
+    phi_mean = ((phi_max + phi_min) / 2.0) % 360
+    theta_mean = (theta_max + theta_min) / 2.0
+    return theta_mean, phi_mean
+
+
+def _minmax_coordinates(vertices):
+    """
+    Return the minimum and maximum values of the latitude and longitude values
+    in the vertex list.
+    """
+    theta_min = np.min(vertices[:, 0])
+    theta_max = np.max(vertices[:, 0])
+    phi_min = np.min(vertices[:, 1])
+    phi_max = np.max(vertices[:, 1])
+    return (theta_min, theta_max), (phi_min, phi_max)
+
+
+def _transform_vertices_plane_parallel(vertices, planet_radius, altitude):
+    """
+    Convert the (lat, lon, elevation) vertices from the initial vertex generation
+    into (x, y, z) values for plane parallel atmosphere geometries.
+
+    Parameters
+    ----------
+
+    vertices : list
+        List of mesh vertices in (lat, lon, elevation) tuples.
```

Comment:
I think this type is inconsistent with the rest of your code. And the usual comment about units: everywhere or nowhere.